contributor-docs/code-change-guide.md

```diff
@@ -0,0 +1,518 @@
+
+* To run a specific test, use the following commands:
+  ```
+  ./gradlew :sdks:java:harness:test --tests org.apache.beam.fn.harness.CachesTest
+  ./gradlew :sdks:java:harness:test --tests *CachesTest
+  ./gradlew :sdks:java:harness:test --tests *CachesTest.testClearableCache
+  ```
+
+* To run tests using IntelliJ, click the ticks to run a whole test class or a specific test. You can set breakpoints to debug the test.
+  <img width="452" alt="image" src="https://github.com/apache/beam/assets/8010435/7ae2a65c-a104-48a2-8bad-ff8c52dd1943">
+
+* These steps don't apply to `sdks:java:core` tests. Instead, invoke the unit tests by using `:runners:direct-java:needsRunnerTest`. Java core doesn't depend on a runner. Therefore, unit tests that run a pipeline require the Direct Runner.
+
+  Reason: Java core does not depend on a runner. For unit test that executes a pipeline, it needs direct runner.
+
+To run integration tests, use the Direct Runner.
+
+## Run integration tests (*IT.java)
+
+Integration tests use [`TestPipeline`](https://github.com/apache/beam/blob/master/sdks/java/core/src/main/java/org/apache/beam/sdk/testing/TestPipeline.java). Set options by using `TestPipelineOptions`.
+
+Integration tests differ from standard pipelines in the following ways:
+* By default, they block on run (on TestDataflowRunner).
+* They have default timeout of 15 minutes.
+* The pipeline options are set in the system property `beamTestPipelineOptions`.
+
+Note the final different, because you need to configure the test by setting `-DbeamTestPipelineOptions=[...]`. This property is where you set the runner to use.
+
+The following example demonstrates how to run an integration test by using the command line. This example includes the options required to run the pipeline on the Dataflow runner:
+```
+-DbeamTestPipelineOptions='["--runner=TestDataflowRunner","--project=mygcpproject","--region=us-central1","--stagingLocation=gs://mygcsbucket/path"]'
+```
+
+### Write an integration test
+
+* To set up a `TestPipeline` object in an integration test, use the following code:
+  ```java
+
+  @Rule public TestPipeline pipelineWrite = TestPipeline.create();
+
+  @Test
+  public void testSomething() {
+    pipeline.apply(...);
+
+    pipeline.run().waitUntilFinish();
+  }
+  ```
+
+* The task that runs the test needs to specify the runner. The following examples demonstrate how to specify the runner:
+  * To run a Google Cloud I/O integration test on the Direct Runner, use `:sdks:java:io:google-cloud-platform:integrationTest`
+  * To run integration tests on the standard Dataflow runner, use `:runners:google-cloud-dataflow-java:googleCloudPlatformLegacyWorkerIntegrationTest`
+  * To run integration test on Dataflow runner v2, use `:runners:google-cloud-dataflow-java:googleCloudPlatformRunnerV2IntegrationTest`
+
+To see how to run your workflow locally, refer to the Gradle command that the GHA workflow runs.
```

Comment:
What does GHA stand for?

Reply:
GitHub Action, will add this abbrev at the first place it occurred